Import aldryn newsblog

arcriley opened the pull request:
This should address remaining issues for Django 3 migration and resolves the merge conflicts with upstream.

arcriley committed:
No longer displays suborg applications menu choices for students, fixes #382

arcriley committed:
Replace previous code with "not user.is_current_year_student" which only triggers is user is currently a student

Commit:
Further work on Django 3 migration

arcriley committed:
Adding migrations for aldryn_categories, aldryn_newsblog, and aldryn_people

arcriley committed:
Merge Terri's changes to the email invite

warthog9 committed:
Merge branch 'master' of https://github.com/python-gsoc/python-blogs into import_aldryn_newsblog

arcriley committed:
Adding aldryn dependencies, removing redundant dependencies

arcriley committed:
Updating templates to use "static" instead of "staticfiles"

arcriley committed:
Now requires djangocms-text-ckeditor version 3.9.1 or above

This fixes a TypeError in version 3.9.0 and below:
 from_db_value() missing 1 required positional argument: 'context'
in djangocms's render_model function

botanicvelious commented on requirements.txt:
```diff
@@ -51,30 +64,12 @@ beautifulsoup4>=4.8.0
-aldryn-translation-tools
-
-#aldryn-newsblog
-python-dateutil
```
This is required for newsblog so it cant be removed

botanicvelious replied:
You have 43 unapplied migration(s). Your project may not work properly until you apply the migrations for app(s): aldryn_categories, aldryn_newsblog, aldryn_people, bootstrap4_alerts, bootstrap4_badge, bootstrap4_card, bootstrap4_carousel, bootstrap4_collapse, bootstrap4_content, bootstrap4_grid, bootstrap4_jumbotron, bootstrap4_link, bootstrap4_listgroup, bootstrap4_media, bootstrap4_picture, bootstrap4_tabs, bootstrap4_utilities, djangocms_googlemap, djangocms_icon.
Run 'python manage.py migrate' to apply them.

botanicvelious reviewed (changes requested):
looks like tests broke when we pulled in newsblog

botanicvelious commented on requirements.txt:
```diff
@@ -51,30 +64,12 @@ beautifulsoup4>=4.8.0
-aldryn-translation-tools
-
-#aldryn-newsblog
-python-dateutil
```
This is required for newsblog so it cant be removed

botanicvelious reviewed (changes requested):
You have 43 unapplied migration(s). Your project may not work properly until you apply the migrations for app(s): aldryn_categories, aldryn_newsblog, aldryn_people, bootstrap4_alerts, bootstrap4_badge, bootstrap4_card, bootstrap4_carousel, bootstrap4_collapse, bootstrap4_content, bootstrap4_grid, bootstrap4_jumbotron, bootstrap4_link, bootstrap4_listgroup, bootstrap4_media, bootstrap4_picture, bootstrap4_tabs, bootstrap4_utilities, djangocms_googlemap, djangocms_icon.
Run 'python manage.py migrate' to apply them.

arcriley committed:
Re-adding python-dateutil to requirements.txt

arcriley committed:
Correcting ValueError: The ADMINS setting must be a list of 2-tuples.

arcriley committed:
Added django-haystack and aldryn-search to the requirements list
Added required basic configuration for haystack
Require haystack 3.0 beta2 which works with Django 3
Removed mail_admins from the default gsoc/settings.py ERROR_HANDLERS

pep8speaks commented:
Hello @arcriley! Thanks for updating this PR. 





* In the file [`gsoc/settings.py`](https://github.com/python-gsoc/python-blogs/blob/eb1ede6f0a2d96338c960d20bf4d3001b44df413/gsoc/settings.py):

> [Line 256:1](https://github.com/python-gsoc/python-blogs/blob/eb1ede6f0a2d96338c960d20bf4d3001b44df413/gsoc/settings.py#L256): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '



##### Comment last updated at 2020-06-27 07:37:38 UTC

arcriley committed:
Changed the test_str_malicious test to expect hexadecimal escaped quotes
instead of decimal escaped (&#39; becomes &#x27;) so it passes.

arcriley committed:
Updating aldryn_categories and aldryn_people tests for Django 3,
specifically "import django.utils.six" becomes "import six"
